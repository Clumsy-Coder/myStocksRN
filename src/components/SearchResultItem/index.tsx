import React from 'react';
import { useNavigation } from '@react-navigation/native';
import { ListItem, Text, Left, Right, Icon } from 'native-base';

import { DataDomain } from '@redux/Stocks/Types';
import { NavigationRoutePath, StockDetailsNavigationProp } from '@routes/Types';

interface Props {
  data: DataDomain.Symbols;
  /**
   * function used for fetching Stock Quote. Check src/Containers/SearchBar/index.tsx .
   * The stock quote needs to be fetched before rendering data in StockDetails screen.
   * Or it will crash.
   */
  onPress: () => void;
}

const StockSearchResultItem: React.FC<Props> = (props: Props) => {
  const { data, onPress: onPressHandler } = props;
  const navigation = useNavigation<StockDetailsNavigationProp>();

  return (
    <ListItem
      onPress={(): void => {
        onPressHandler();
        navigation.navigate(NavigationRoutePath.StockDetails, { symbol: data.symbol });
      }}
      key={`search-result-item-${data.symbol}`}
    >
      <Left>
        <Text>
          {data.symbol}: {data.name}
        </Text>
      </Left>
      <Right>
        <Icon name='arrow-forward' type='MaterialIcons' />
      </Right>
    </ListItem>
  );
};

export default StockSearchResultItem;
