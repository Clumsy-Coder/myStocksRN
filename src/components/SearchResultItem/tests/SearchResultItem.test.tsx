import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

import SearchResultItem from 'src/components/SearchResultItem';
import * as testdata from 'jest.testdata';

describe('<SearchResultItem />', () => {
  const props = {
    data: testdata.symbolsMetadata1,
    onPress: jest.fn(),
  };

  it('Render correctly', () => {
    const wrapper = shallow(<SearchResultItem {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('Handles onPress event', () => {
    const renderProps = { ...props };
    const wrapper = shallow(<SearchResultItem {...renderProps} />);

    wrapper.props().onPress();

    expect(renderProps.onPress).toBeCalledTimes(1);
    expect(wrapper).toMatchSnapshot();
  });
});
