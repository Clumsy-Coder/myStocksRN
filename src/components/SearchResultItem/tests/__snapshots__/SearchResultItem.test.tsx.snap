// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResultItem /> Handles onPress event 1`] = `
<Styled(ListItem)
  key="search-result-item-IBM"
  onPress={[Function]}
>
  <Styled(Left)>
    <Styled(Text)>
      IBM
      : 
      International Business Machines Corporation
    </Styled(Text)>
  </Styled(Left)>
  <Styled(Right)>
    <Styled(Icon)
      name="arrow-forward"
      type="MaterialIcons"
    />
  </Styled(Right)>
</Styled(ListItem)>
`;

exports[`<SearchResultItem /> Render correctly 1`] = `
<Styled(ListItem)
  key="search-result-item-IBM"
  onPress={[Function]}
>
  <Styled(Left)>
    <Styled(Text)>
      IBM
      : 
      International Business Machines Corporation
    </Styled(Text)>
  </Styled(Left)>
  <Styled(Right)>
    <Styled(Icon)
      name="arrow-forward"
      type="MaterialIcons"
    />
  </Styled(Right)>
</Styled(ListItem)>
`;
